presto-client/src/main/java/com/facebook/presto/client/ClientSession.java

```diff
@@ -134,6 +137,14 @@ public ClientSession(
             checkArgument(charsetEncoder.canEncode(entry.getKey()), "Credential name is not US_ASCII: %s", entry.getKey());
             checkArgument(charsetEncoder.canEncode(entry.getValue()), "Credential value is not US_ASCII: %s", entry.getValue());
         }
+
+        // verify the extra credentials are valid
+        for (Entry<String, String> entry : customHeaders.entrySet()) {
+            checkArgument(!entry.getKey().isEmpty(), "Credential name is empty");
+            checkArgument(entry.getKey().indexOf('=') < 0, "Credential name must not contain '=': %s", entry.getKey());
+            checkArgument(charsetEncoder.canEncode(entry.getKey()), "Credential name is not US_ASCII: %s", entry.getKey());
+            checkArgument(charsetEncoder.canEncode(entry.getValue()), "Credential value is not US_ASCII: %s", entry.getValue());
```

Comment:
checkArguments will throw IllegalArgumentException
I think we will have to wrap them as Presto user errors.

Reply:
These checks seem to be repeated . 
Maybe have a wrapper method called `isValid(Map<String, String> properties)` to run the same check everywhere.

On a side note, curious why the resourceEstimates doesn't need a valid value check